Gems/ROS2/Code/Source/RobotImporter/ROS2RobotImporterEditorSystemComponent.cpp

```diff
@@ -61,4 +75,92 @@ namespace ROS2
+            return false;
+        }
+        if (Utils::IsFileXacro(filePath))
+        {
+            AZ_Warning("ROS2EditorSystemComponent", false, "XACRO is not supported");
+            return false;
+        }
+
+        urdf::ModelInterfaceSharedPtr parsedUrdf = UrdfParser::ParseFromFile(filePath);
+        if (!parsedUrdf)
+        {
+            AZ_Warning("ROS2EditorSystemComponent", false, "URDF parsing failed");
+            const auto log = UrdfParser::GetUrdfParsingLog();
+            AZ_Warning("ROS2EditorSystemComponent", false, "%s", log.c_str());
+            return false;
+        }
+
+        auto collidersNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), false, true);
+        auto visualNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), true, false);
+        auto meshNames = Utils::GetMeshesFilenames(parsedUrdf->getRoot(), true, true);
+        AZStd::shared_ptr<Utils::UrdfAssetMap> urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>();
+        if (importAssetWithUrdf)
+        {
+            urdfAssetsMapping = AZStd::make_shared<Utils::UrdfAssetMap>(
+                Utils::CopyAssetForURDFAndCreateAssetMap(meshNames, filePath, collidersNames, visualNames));
+        }
+        bool allAssetProcessed = false;
+        do
+        {
+            allAssetProcessed = true;
+            for (const auto& [name, asset] : *urdfAssetsMapping)
+            {
+                auto sourceAssetFullPath = asset.m_availableAssetInfo.m_sourceAssetGlobalPath;
+                if (sourceAssetFullPath.empty())
+                {
+                    AZ_Warning("ROS2EditorSystemComponent", false, "asset %s is has no sourceAssetFullPath", name.c_str());
+                    continue;
+                }
+                using namespace AzToolsFramework;
+                using namespace AzToolsFramework::AssetSystem;
+                AZ::Outcome<AssetSystem::JobInfoContainer> result = AZ::Failure();
+                AssetSystemJobRequestBus::BroadcastResult(
+                    result, &AssetSystemJobRequestBus::Events::GetAssetJobsInfo, sourceAssetFullPath, true);
+                JobInfoContainer& allJobs = result.GetValue();
+                for (const JobInfo& job : allJobs)
+                {
+                    if (job.m_status == JobStatus::Queued || job.m_status == JobStatus::InProgress)
+                    {
+                        AZ_Printf("ROS2EditorSystemComponent", "asset %s is being processed", sourceAssetFullPath.c_str());
+                        allAssetProcessed = false;
+                    }
+                    else
+                    {
+                        AZ_Printf("ROS2EditorSystemComponent", "asset %s is done", sourceAssetFullPath.c_str());
+                    }
+                }
+            }
+
+            if (allAssetProcessed)
+            {
+                AZ_Printf("ROS2EditorSystemComponent", "All assets processed");
+            }
+        } while (!allAssetProcessed);
```

Comment:
The stop condition here is not fully addressed. We have an infinite loop if the asset system does not update the job status which can happen for some reasons. This needs a timeout or another kind of stop condition to handle this.